src/amplitude/client.py

```diff
@@ -163,7 +163,9 @@ def remove(self, plugin: Plugin):
         self.__timeline.remove(plugin)
         return self
 
-    def shutdown(self):
+    def shutdown(self, grace_time_milliseconds=0):
         """Shutdown the client instance, not accepting new events, flush all events in buffer"""
         self.configuration.opt_out = True
         self.__timeline.shutdown()
+        if grace_time_milliseconds > 0:
```

Comment:
Thanks for the quick update! Not sure whether we should add this here, as if `shutdown` is exposed as public API, user can do the `time.sleep` if they need. Do we consider to use async tasks for the worker? https://docs.python.org/3/library/asyncio-task.html.

Reply:
The current worker use threading. Switch to coroutine may need some work to re-design and implement the retry logic. shutdown() is a public interface suppose to be called before application exit. I have run some tests. Add ether flush or shutdown after tracking will stop the error of `can not schedule future`. Let me ask customer if they added these call.

```
from amplitude import *


client = Amplitude('')

client.track(BaseEvent(
    event_type='Python Event',
    user_id='datamonster@gmail.com',
    ip='127.0.0.1',
    event_properties={
        'keyString': 'valueString',
        'keyInt': 11,
        'keyBool': True
    }
))

# client.flush()
client.shutdown()```

Reply:
I see, I thought this fix is for the other issue (timing issue). I cannot reproduce the error `can not schedule future` in both Python 3.9 and 3.10. If adding this can fix the issue, I am good with change. Just to double check, will this be stable with the usage of `sleep`?

Reply:
@bohan-amplitude Can you please first enable py37-p311 pipelines, so we can see what really happens with different python versions, running only on py36 is insane, that version pinned for the fjords long time ago.

Reply:
Hi @ssbarnea, just shared open another pull request with workaround for this issue at https://github.com/amplitude/Amplitude-Python/pull/43, feel free to take a look and let us know your thoughts. Seems that passes the test on py3.6 to py3.11.